src/services/completions.ts

```diff
@@ -3663,71 +3664,130 @@ function getCompletionData(
+                        omitTrailingSemicolon: true,
+                    });
+                    const origin: SymbolOriginInfoComputedPropertyName = {
+                        kind: getNullableSymbolOriginInfoKind(SymbolOriginInfoKind.SymbolMemberNoExport) | SymbolOriginInfoKind.ComputedPropertyName,
+                        symbolName: printer.printNode(EmitHint.Unspecified, node, contextToken.getSourceFile()),
+                    };
+                    symbolToOriginInfoMap[index] = origin;
                 }
             }
-            else if (preferences.includeCompletionsWithInsertText) {
-                if (firstAccessibleSymbolId && seenPropertySymbols.has(firstAccessibleSymbolId)) {
-                    return;
+            else {
+                const fileName = isExternalModuleNameRelative(stripQuotes(moduleSymbol.name)) ? getSourceFileOfModule(moduleSymbol)?.fileName : undefined;
+                const { moduleSpecifier } = (importSpecifierResolver ||= codefix.createImportSpecifierResolver(sourceFile, program, host, preferences)).getModuleSpecifierForBestExportInfo(
+                    [{
+                        exportKind: ExportKind.Named,
+                        moduleFileName: fileName,
+                        isFromPackageJson: false,
+                        moduleSymbol,
+                        symbol: firstAccessibleSymbol,
+                        targetFlags: skipAlias(firstAccessibleSymbol, typeChecker).flags,
+                    }],
+                    position,
+                    isValidTypeOnlyAliasUseSite(location),
+                ) || {};
+
+                if (moduleSpecifier) {
+                    const origin: SymbolOriginInfoResolvedExport = {
+                        kind: getNullableSymbolOriginInfoKind(SymbolOriginInfoKind.SymbolMemberExport),
+                        moduleSymbol,
+                        isDefaultExport: false,
+                        symbolName: firstAccessibleSymbol.name,
+                        exportName: firstAccessibleSymbol.name,
+                        fileName,
+                        moduleSpecifier,
+                    };
+                    symbolToOriginInfoMap[index] = origin;
                 }
-                addSymbolOriginInfo(symbol);
-                addSymbolSortInfo(symbol);
-                symbols.push(symbol);
             }
         }
-        else {
+
+        function addLiteralSymbol() {
             addSymbolOriginInfo(symbol);
             addSymbolSortInfo(symbol);
             symbols.push(symbol);
         }
 
+        /** Combines a property access expression (from the completion location to the left-most name in the computed property expression)
+         * and the computed property expression itself to return an expression to access the computed property from the completion location.
+         * Assumes that the left-most symbol is accessible so there should always be a way to access the computed property symbolically. */

```

Comment:
That comment was a bit confusing to me. I think having a comment like the one you left above, with an example of what each parameter is and what we want to do, e.g. `x.y`, `y`'s symbol, `m`'s symbol, would help clarify it.